### ISSUE TITLE
IDE crash

### ISSUE THREAD

**Issue by HydrogenC** [beefytech/Beef#104]
Description: 
```
Executable: C:\Program Files\BeefLang\bin\BeefIDE.exe
File Version: 0.42
Product Version: c2ee40181288ee486b179098e62da3494b49ee65

Logical Address: 0000:'00000000

2b'445ff160 '666c6553 0000:'00000000 
    Params: 1bf'98728ca0 1bf'98728ca0 1bf'98928650 1bf'98728ce8

2b'445ff370 1bf'8afbf8b5 IDEHelper64.dll BfParser_SetCompleteParse+49175
    Params: 1bf'93188020 101'00000000 '00000004 1bf'00000000

2b'445ff660 1bf'8afb4f24 IDEHelper64.dll BfParser_SetCompleteParse+3E7E4
    Params: 1bf'93188020 1bf'98eff600 2b'445ff600 1bf'935a17f0

2b'445ff710 1bf'8afc2a3b IDEHelper64.dll BfCompiler_ClassifySource+1FB
    Params: 1bf'935a17f8 1bf'9cc95820 1bf'934fea80 1bf'9cc3bac0

2b'445ff760 7ff6'7e43e804 0001:'0004d804 BeefIDE.exe
    Params: 2b'445ff840 7ff6'7e521342 '00000d03 1bf'9cc3bac0

2b'445ff920 7ff6'7e5213e3 0001:'001303e3 BeefIDE.exe
    Params: '00000000 7ff9'fceac387 '00000000 'ffffffff

2b'445ff990 7ff6'7e49b447 0001:'000aa447 BeefIDE.exe
    Params: '00000000 1bf'8ad10d80 '00000000 1bf'8ad10d80

2b'445ffab0 7ff6'7e5e718c 0001:'001f618c BeefIDE.exe
    Params: '00000000 '00000000 '00000000 '00000002

2b'445ffaf0 7ff6'7e5e7410 0001:'001f6410 BeefIDE.exe
    Params: '00000000 '00000000 '00000000 '00000604

2b'445ffb20 7ff9'e7da3dcd Beef042RT64.dll unsigned long bf::System::Threading::Thread::GetProcessDefaultStackSize(void)+9D
    Params: 1bf'92f75120 '00000000 '00000000 '00000000

2b'445ffb50 7ff9'fe057bd4 KERNEL32.DLL BaseThreadInitThunk+14
    Params: '00000000 '00000000 '00000000 '00000000

2b'445ffbd0 7ff9'ff2eced1 ntdll.dll RtlUserThreadStart+21
    Params: '00000000 '00000000 '00000000 '00000000


RAX:1bf'92f523c0 RBX:1bf'98728ca0 RCX:1bf'934ce1b0 RDX:'00000000 RSI:1bf'9842fd50 RDI:'00000000
R8:1bf'931cf620 R9:'00000000 R10:1bf'8c9e80f4 R11:2b'445ff1e3
R12:'00000000 R13:1bf'98a60970 R14:'00000000 R15:'00000000
EIP:'666c6553 ESP:2b'445ff168 EBP:2b'445ff270
CS:0033 SS:002B DS:002B ES:002B FS:0053 GS:002B
Flags:'00010297
```

Crash dump: 
[ide-crash.zip](https://github.com/beefytech/Beef/files/4393260/ide-crash.zip)

**Comment by HydrogenC** [comment]
I have defined a delegate: 
`delegate void(T item) mDelegate;`
When I want to call it like: 
`mDelegate(...);`
It crashes when I type the parentheses after mDelegate.

**Comment by HydrogenC** [comment]
Another crash: 
```
Crash minidump saved as C:\Program Files\BeefLang\bin\CrashDumps\BeefIDE_20200327_202354.dmp

Exception: Access Violation (code 0xc0000005) at address 23d'cca70eca in thread 270
Executable: C:\Program Files\BeefLang\bin\BeefIDE.exe
File Version: 0.42
Product Version: c2ee40181288ee486b179098e62da3494b49ee65

Logical Address: 0001:'0015feca

64'efcfdd00 23d'cca70eca IDEHelper64.dll BfpFindFileData_GetTime_LastWrite+392BA
    Params: 23d'd4cc50d0 23d'dab7b160 23d'00000006 '00000000

64'efcfe360 23d'ccb5d218 IDEHelper64.dll Breakpoint_GetPendingHotBindIdx+F368
    Params: 23d'd95c2e00 23d'd95c2e18 '00000000 '00000000

64'efcfe3b0 23d'ccb6669d IDEHelper64.dll Breakpoint_GetPendingHotBindIdx+187ED
    Params: 23d'ce569df0 64'efcfeb10 23d'da9e9460 23d'00000001

64'efcfe550 23d'ccb651fa IDEHelper64.dll Breakpoint_GetPendingHotBindIdx+1734A
    Params: 64'efcfe600 23d'd95c3c01 23d'00000002 64'dab79c00

64'efcff0e0 23d'ccad9cb3 IDEHelper64.dll BfpFindFileData_GetTime_LastWrite+A20A3
    Params: fffffdcd'ffffff00 201'fffffdbe 23d'dacdad00 23d'da9e9400

64'efcff3d0 23d'ccb34b3c IDEHelper64.dll BfParser_SetCompleteParse+3E3FC
    Params: 23d'd4e9a730 23d'daab8300 64'efcff400 23d'da597c10

64'efcff480 23d'ccb42a3b IDEHelper64.dll BfCompiler_ClassifySource+1FB
    Params: 23d'da597c18 23d'dacd9000 23d'd50d8eb0 23d'db4c2e70

64'efcff4d0 7ff6'7e43e804 0001:'0004d804 BeefIDE.exe
    Params: 64'efcff5b0 7ff6'7e521342 '00000e03 23d'db4c2e70

64'efcff690 7ff6'7e5213e3 0001:'001303e3 BeefIDE.exe
    Params: '00000000 7ff9'fceac387 '00000000 'ffffffff

64'efcff700 7ff6'7e49b447 0001:'000aa447 BeefIDE.exe
    Params: '00000000 23d'cc891a90 '00000000 23d'cc891a90

64'efcff820 7ff6'7e5e718c 0001:'001f618c BeefIDE.exe
    Params: '00000000 '00000000 '00000000 '00000002

64'efcff860 7ff6'7e5e7410 0001:'001f6410 BeefIDE.exe
    Params: '00000000 '00000000 '00000000 '000006c8

64'efcff890 7ff9'e6a13dcd Beef042RT64.dll unsigned long bf::System::Threading::Thread::GetProcessDefaultStackSize(void)+9D
    Params: 23d'd5202ff0 '00000000 '00000000 '00000000

64'efcff8c0 7ff9'fe057bd4 KERNEL32.DLL BaseThreadInitThunk+14
    Params: '00000000 '00000000 '00000000 '00000000

64'efcff940 7ff9'ff2eced1 ntdll.dll RtlUserThreadStart+21
    Params: '00000000 '00000000 '00000000 '00000000


RAX:'00000000 RBX:23d'd95c2f50 RCX:64'efcfd980 RDX:64'efcfd988 RSI:23d'da686060 RDI:'00000000
R8:'00000002 R9:23d'cc910000 R10:64'efcfd988 R11:64'efcfd9d0
R12:23d'da25bc50 R13:'00000006 R14:23d'd4cc50d0 R15:23d'd95c2e60
EIP:23d'cca70eca ESP:64'efcfd910 EBP:64'efcfda10
CS:0033 SS:002B DS:002B ES:002B FS:0053 GS:002B
Flags:'00010246
```

Crash dump: 
[crash_ide_2.zip](https://github.com/beefytech/Beef/files/4393467/crash_ide_2.zip)

**Comment by bfiete** [comment]
The delegate crash is fixed here https://github.com/beefytech/Beef/commit/6f975686f77259afb7315c994a42777807650491, will be in 0.42.2.

**Comment by HydrogenC** [comment]
Thanks

**Comment by damianday** [comment]
@HydrogenC I cant get it to crash with 0.42.2 after typing mDelegate... this was my test code below.

```
public class Temp<T>
{
	delegate void(T item) mDelegate;

	public void fire(T value)
	{
		mDelegate(value);
	}
}
```

**Comment by HydrogenC** [comment]
@damianday The first bug is already fixed in 0.42.2.
